### PR TITLE
Speed up a little

### PR DESCRIPTION
**Pull request by aarmey**
Sometimes the data import can be a little slow. I've set the results to cache as a quick fix. Longer-term, it would be much more efficient to setup the pandas table (get the `split` steps out of the way), and then use `xarray.stack` to form the tensor. Indexing pandas tables in loops can be very slow.

### CODE REVIEW

**Comment by cyrillustan** [comment]
Can you give me an example of the function you are referring to? I can't find the function you said that can convert a pandas df into a 3D xarray.

**Comment by aarmey** [comment]
I think you make a MultiIndex with the dimensions you want, then you use `stack()`: https://docs.xarray.dev/en/stable/user-guide/reshaping.html#stack-and-unstack

**Comment by cyrillustan** [comment]
Does this look right? It seems to speed up comparing with the older versions

File: tensordata/kaplonekVaccine2022/Luminex-functional-assay.csv [hunk 1]
```diff
@@ -17,7 +17,7 @@ Male,32.57573165,BNT162b2,320.5,123.1,150.94899,114.95,109.08,1.100135,0.58529,1
...
-Male,34.56619004,mRNA-1273,433.5,95.26,138.91746,114.25,87.7,1.91844,0.9065,1.663825,1.38003,39.75,8.625,55.35,4798.5,1661,245.5,3309,4395.5,2027.5,2247,1722,49.8,25.4,25.4,49.8,49.8,74.2,135,49.8,2748,1234,342.5,1954.5,3675.5,2564.5,2845,2418,330.5,184,98.7,221,416,489,941,819,294,245,98.7,270,306,379.5,465,416,30323,18494.5,6349,23878,27882,22120,25525.5,22510.5,26526.5,15027.5,501.5,22144.5,24073,19788.5,25892,22437,25354.5,16431.5,6458.5,20289,24219.5,18409,20850.5,18128.5,25318.5,16139,2259.5,20386.5,23792,19495.5,20582,18360
+Male,34.56619003,mRNA-1273,433.5,95.26,138.91746,114.25,87.7,1.91844,0.9065,1.663825,1.38003,39.75,8.625,55.35,4798.5,1661,245.5,3309,4395.5,2027.5,2247,1722,49.8,25.4,25.4,49.8,49.8,74.2,135,49.8,2748,1234,342.5,1954.5,3675.5,2564.5,2845,2418,330.5,184,98.7,221,416,489,941,819,294,245,98.7,270,306,379.5,465,416,30323,18494.5,6349,23878,27882,22120,25525.5,22510.5,26526.5,15027.5,501.5,22144.5,24073,19788.5,25892,22437,25354.5,16431.5,6458.5,20289,24219.5,18409,20850.5,18128.5,25318.5,16139,2259.5,20386.5,23792,19495.5,20582,18360
```
Comment (aarmey) [inline tensordata/kaplonekVaccine2022/Luminex-functional-assay.csv:20]: How did this change?

**Review by aarmey — COMMENTED** [review]
Yeah! This is also more concise in the end and so, with comments, could be more readable.

File: tensordata/kaplonekVaccine2022/Luminex-functional-assay.csv (hunk 1 above)
Comment (cyrillustan) [reply tensordata/kaplonekVaccine2022/Luminex-functional-assay.csv:20]: There’s no sample number in this so we use age as identifier. Two people have identical age so I made a small change to make it unique.

**Review by cyrillustan — APPROVED** [review]
I think this all works well. Ready to merge?

**Pull request merged**